In [30]:
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
import scipy.stats as stats

In [15]:
data_name = "cumulative-co-emissions.csv"
data_table = pd.read_csv(data_name)
data_table.head()

YR           CO2  land_ocean_temp_index  land_ocean_temp_index_smoothed
0  1880  5.596738e+10                  -0.19                           -0.11
1  1881  5.861068e+10                  -0.10                           -0.14
2  1882  6.139991e+10                  -0.11                           -0.18
3  1883  6.436377e+10                  -0.19                           -0.21
4  1884  6.736057e+10                  -0.28                           -0.24

In [20]:
data_table.set_index('YR').head()


CO2  land_ocean_temp_index  land_ocean_temp_index_smoothed
YR                                                                       
1880  5.596738e+10                  -0.19                           -0.11
1881  5.861068e+10                  -0.10                           -0.14
1882  6.139991e+10                  -0.11                           -0.18
1883  6.436377e+10                  -0.19                           -0.21
1884  6.736057e+10                  -0.28                           -0.24

In [29]:

first_period = data_table.loc[data_table["YR"] < 1950]
first_period

YR           CO2  land_ocean_temp_index  land_ocean_temp_index_smoothed
0   1880  5.596738e+10                  -0.19                           -0.11
1   1881  5.861068e+10                  -0.10                           -0.14
2   1882  6.139991e+10                  -0.11                           -0.18
3   1883  6.436377e+10                  -0.19                           -0.21
4   1884  6.736057e+10                  -0.28                           -0.24
5   1885  7.037772e+10                  -0.31                           -0.26
6   1886  7.344079e+10                  -0.32                           -0.27
7   1887  7.665672e+10                  -0.35                           -0.27
8   1888  8.022013e+10                  -0.18                           -0.27
9   1889  8.378102e+10                  -0.11                           -0.26
10  1890  8.765868e+10                  -0.37                           -0.25
11  1891  9.171293e+10                  -0.23                           -0.26
12  1892  9.579509e+10                  -0.26                           -0.27
13  1893  9.982440e+10                  -0.31                           -0.26
14  1894  1.039947e+11                  -0.31                           -0.24
15  1895  1.084129e+11                  -0.22                           -0.22
16  1896  1.129752e+11                  -0.11                           -0.21
17  1897  1.177529e+11                  -0.12                           -0.18
18  1898  1.227916e+11                  -0.27                           -0.17
19  1899  1.282890e+11                  -0.18                           -0.18
20  1900  1.340754e+11                  -0.09                           -0.20
21  1901  1.400540e+11                  -0.15                           -0.24
22  1902  1.461892e+11                  -0.28                           -0.26
23  1903  1.528903e+11                  -0.38                           -0.29
24  1904  1.596564e+11                  -0.48                           -0.32
25  1905  1.668692e+11                  -0.28                           -0.34
26  1906  1.744755e+11                  -0.22                           -0.36
27  1907  1.830220e+11                  -0.39                           -0.38
28  1908  1.912235e+11                  -0.43                           -0.39
29  1909  1.997697e+11                  -0.47                           -0.40
..   ...           ...                    ...                             ...
40  1920  3.062946e+11                  -0.26                           -0.26
41  1921  3.153437e+11                  -0.18                           -0.25
42  1922  3.248354e+11                  -0.28                           -0.24
43  1923  3.356501e+11                  -0.25                           -0.23
44  1924  3.464513e+11                  -0.26                           -0.22
45  1925  3.573462e+11                  -0.21                           -0.21
46  1926  3.681205e+11                  -0.09                           -0.21
47  1927  3.798360e+11                  -0.20                           -0.20
48  1928  3.915621e+11                  -0.19                           -0.18
49  1929  4.041675e+11                  -0.34                           -0.18
50  1930  4.157382e+11                  -0.14                           -0.18
51  1931  4.260258e+11                  -0.09                           -0.18
52  1932  4.352552e+11                  -0.16                           -0.18
53  1933  4.449644e+11                  -0.28                           -0.17
54  1934  4.555032e+11                  -0.14                           -0.16
55  1935  4.665624e+11                  -0.20                           -0.15
56  1936  4.787012e+11                  -0.16                           -0.12
57  1937  4.917059e+11                  -0.04                           -0.08
58  1938  5.039215e+11                  -0.03                           -0.03
59  1939  5.167875e+11      

In [32]:
second_period = data_table.loc[data_table["YR"] >= 1950]
second_period.head()

YR           CO2  land_ocean_temp_index  land_ocean_temp_index_smoothed
70  1950  6.781871e+11                  -0.18                           -0.08
71  1951  6.962909e+11                  -0.07                           -0.07
72  1952  7.146129e+11                   0.01                           -0.07
73  1953  7.334089e+11                   0.07                           -0.08
74  1954  7.524743e+11                  -0.14                           -0.07

In [33]:
stats.ttest_ind(first_period['land_ocean_temp_index_smoothed'], second_period['land_ocean_temp_index_smoothed'], equal_var=False)

Ttest_indResult(statistic=-12.236812059147972, pvalue=7.025285678956423e-21)

In [34]:
stats.ttest_ind(first_period['CO2'], second_period['CO2'], equal_var=False)

Ttest_indResult(statistic=-13.06026071409178, pvalue=2.3732541362006872e-20)

Based on P value from T test, we can see that there is a significant difference in temperature index over 2 time periods: 1880 to 1950 vs 1950 to 2019. We notice same pattern for CO2 as well.